src/java.base/share/classes/java/lang/runtime/ObjectMethods.java

```diff
@@ -317,19 +319,28 @@ private static MethodHandle makeToString(Class<?> receiverClass,
      * @param recordClass  the record class hosting the record components
      * @param names        the list of component names, joined into a string
      *                     separated by ";", or the empty string if there are no
-     *                     components. Maybe be null, if the {@code methodName}
-     *                     is {@code "equals"} or {@code "hashCode"}.
+     *                     components. This parameter is ignored if the {@code methodName}
+     *                     parameter is {@code "equals"} or {@code "hashCode"}
      * @param getters      method handles for the accessor methods for the components
      * @return             a call site if invoked by indy, or a method handle
      *                     if invoked by a condy
      * @throws IllegalArgumentException if the bootstrap arguments are invalid
      *                                  or inconsistent
+     * @throws NullPointerException if any argument but {@code lookup} is {@code null},
+     *                              in the case of the {@code getters} argument, its
+     *                              contents cannot be {@code null} either
      * @throws Throwable if any exception is thrown during call site construction
      */
     public static Object bootstrap(MethodHandles.Lookup lookup, String methodName, TypeDescriptor type,
                                    Class<?> recordClass,
                                    String names,
                                    MethodHandle... getters) throws Throwable {
+        requireNonNull(methodName);
+        requireNonNull(type);
+        requireNonNull(recordClass);
+        requireNonNull(names);
+        requireNonNull(getters);
+        Arrays.stream(getters).forEach(Objects::requireNonNull);
         MethodType methodType;
         if (type instanceof MethodType)
```

Comment:
Since you are modifying this file, do you mind taking Jesper's suggestion [1] posted in another PR to use pattern matching.

```suggestion
        if (type instanceof MethodType mt)
             methodType = mt;
```

[1] https://github.com/openjdk/valhalla/pull/528#discussion_r688100918

Reply:
sure I will do this before pushing, thanks